pymeasure/instruments/thyracont/smartline_v2.py

```diff
@@ -0,0 +1,505 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from enum import IntEnum
+
+from pymeasure.instruments import Instrument, Channel, validators
+from pyvisa.constants import Parity, StopBits
+
+import pyvisa as pv
+from pyvisa import VisaIOError  # noqa: F401 for dependencies
+
+
+def calculateChecksum(message):
+    """Calculate the checksum for string `message`."""
+    value = 0
+    for i in range(len(message)):
+        value += ord(message[i])
+    return chr(value % 64 + 64)
+
+
+def compose_data(value):
+    """Generate a string with the length of `value` and the `value` itself afterwards.
+
+    :param value: Value to send to the device.
+    """
+    value = f"{value}"
+    return f"{len(value):02}{value}"
+
+
+source = {
+    'combination': 0,
+    'pirani': 1,
+    'piezo': 2,
+    'hotCathode': 3,
+    'coldCathode': 4,
+    'ambient': 6,
+    'relative': 7,
+}
+
+
+class Sources(IntEnum):
+    COMBINATION = 0
+    PIRANI = 1
+    PIEZO = 2
+    HOT_CATHODE = 3
+    COLD_CATHODE = 4
+    AMBIENT = 6
+    RELATIVE = 7
+
+
+gas_factor = Channel.control(
+    "0C{ch}00", "0C{ch}%s", "Control the gas correction factor.",
+    values=(0.2, 8),
+    validator=validators.strict_range,
+    set_process=compose_data,
+)
+
+
+class SensorChannel(Channel):
+    """Generic channel for individual pressure sensors of a Transmitter."""
+
+    _id = -1  # obligatory channel number, define in channel types
+
+    def __init__(self, parent, id):
+        if id is None:
+            id = self._id
+        assert id == self._id, f"Pirani ID has to be {self._id}."
+        super().__init__(parent, id=self._id)
+
+    pressure = Channel.measurement(
+        "0M{ch}00", """Get the current pressure in mbar.""",
+        preprocess_reply=lambda r: r.replace("UR", "0").replace("OR", "inf"),
+    )
+
+
+class Pirani(SensorChannel):
+    """Pirani sensor channel.
+
+    A Pirani sensor measures the heat transfer in the gas. The reading depends on the gas type.
+    The sensor reading is linear to the real pressure below some threshold, around 1 mbar.
+    You may control a gas factor with :attr:`gas_factor`.
+    """
+
+    _id = 1
```

Comment:
```suggestion
    _id = source ['pirani'] # or use enum ?
```
You defined symbolic names above and here you use the literal number. I don't understand or I missed your point. Is it intentional?

Reply:
You're right, that we have the enum. Thanks for hinting at it.